cmd/nri-ecs_test.go

```diff
@@ -43,21 +40,62 @@ func TestIntegrationPublish(t *testing.T) {
 		os.Stdout = stdout
 	}()
 
-	main()
+	cases := map[string]struct {
+		response       string
+		endpointEnvVar string
+	}{
+		"from_v3_metadata_endpoint": {
+			response:       v3,
+			endpointEnvVar: metadata.ContainerMetadataEnvVar,
+		},
+		"from_v4_metadata_endpoint": {
+			response:       v4,
+			endpointEnvVar: metadata.ContainerMetadataV4EnvVar,
+		},
+	}
+	for testCaseName, testData := range cases {
+		testData := testData
 
-	// Read the integration output from the captured stdout
-	b := make([]byte, 1024)
-	n, err := readerOut.Read(b)
-	if err != nil {
-		t.Fatal(err)
+		t.Run(testCaseName, func(t *testing.T) {
+			ts := metadataServer(t, testData.response)
+			t.Setenv(testData.endpointEnvVar, ts.URL)
+
+			ecsIntegration, _ := integration.New("com.newrelic.ecs", "0.0.0")
+
+			assert.NoError(t, main.Run(ecsIntegration, args))
+
+			// Read the integration output from the captured stdout
+			b := make([]byte, 1024)
+			n, err := readerOut.Read(b)
+			if err != nil {
+				t.Fatal(err)
+			}
+			var result interface{}
+			err = json.Unmarshal(b[:n], &result)
+			require.NoError(t, err)
+
+			var expected interface{}
+			err = json.Unmarshal([]byte(testdata.IntegrationOutput), &expected)
+			require.NoError(t, err)
+
+			assert.Equal(t, expected, result)
+		})
 	}
-	var result interface{}
-	err = json.Unmarshal(b[:n], &result)
-	require.NoError(t, err)
+}
+
+func metadataServer(t *testing.T, response string) *httptest.Server {
+	testServer := httptest.NewServer(http.HandlerFunc(
```

Comment:
t.Helper() Missing? I am not sure if it is a best practise to always include it in helper functions

Reply:
yes it is thanks i will add it.